lib/sms_signup.rb

```diff
@@ -0,0 +1,32 @@
+class SmsSignup
+  def initialize(user_model:)
+    @user_model = user_model
+  end
+
+  def execute(contact:)
+    phone_number = normalised_phone_number(contact)
+    login_details = user_model.generate(contact: phone_number)
+    notify_params = { login: login_details[:username], pass: login_details[:password] }
```

Comment:
Not sure this could ever be nil, but if it were, and nils some got into the database, it would be a successful authentication?  .fetch instead of [ would raise an error which might be better.

Reply:
Oh sorry no, see this is for Notify.